### DISCUSSION TITLE
View Transitions: able to be off-by-default (rather than fade) and enable per-element?

### DISCUSSION THREAD

**Discussion by JoshuaKGoldberg** [withastro/roadmap#633] [Proposal]
## Summary

Can we have first-class Astro support for View Transitions API that defaults to no animation, rather than `"fade"`?

## Background & Motivation

I just enabled `<ViewTransitions />` in the shared `<head>` component of my site, and it works great - pages fade very smoothly on navigation. Awesome job!

But, I don't want to have the fade animation on the _entire_ page. I'd like to have it only apply to specific elements, and have a more traditional swap for everything else.

It doesn't seem like `<ViewTransitions />` has an option for setting to `"none"` _(or an equivalent)_ by default instead of `"fade"`. But if I don't include `<ViewTransitions />` in my page, then no Astro transitions in the page work at all. Am I missing something?

Edit: I should also note that fading is the default (see the [MDN View Transition docs](https://developer.mozilla.org/en-US/docs/Web/API/View_Transitions_API#creating_a_basic_view_transition) for a primer). So perhaps this request is really for more seamlessly customizing the default/root transition?

## Goals

Either:

* Add an option / API for pages to switch their default view transition, OR
* A docs mention for folks like me about why this isn't doable? 

## Example

```jsx
<ViewTransitions default="none" />
```

Edit: or...

```jsx
import { fade } from 'astro:transitions';

<ViewTransitions default={fade({ duration: '0.1s' })} />
```

**Comment by natemoo-re** [comment]
This can be accomplished by adding the following declaration to your CSS:

```css
::view-transition-group(root) {
  animation: none;
}
```

It might be worth adding this to the `ViewTransitions` component, but I also think there's benefit in having users learn to use the platform API for this.

**Comment by matthewp** [comment]
We've been talking about this. Might be a good default to change to.
